File: gin-server/gin-blog-server/utils/errmsg/errmsg.go
package errmsg

const (
	SUCCESS = 200
	ERROR   = 500

	// JWT中间件
	TokenNotExist       = 1001
	TokenInvalid        = 1002
	TokenParseFailed    = 1003
	TokenGenerateFailed = 1004

	// 2FA
	GenerateSecretFailed = 3001
	CodeError            = 3002

	// QRcode
	GenerateQRFailed = 4001

	// 管理员用户
	UserInfoError       = 2001
	UserExist           = 2002
	UserNotExist        = 2003
	UserNotValidate     = 2004
	CheckUserFailed     = 2005
	LoginError          = 2006
	UidError            = 2007
	PasswordError       = 2008
	CantLogin           = 2009
	ResetPasswordFailed = 2010
	QueryUserFailed     = 2011

	// 邮件发送
	SendEmailFailed   = 5001
	ValidateCodeError = 5002

	// 博客设置
	BlogNotSet    = 6001
	BlogSetFailed = 6002

	// 文章管理
	ArticleUpdateFailed = 7001
	ArticleQueryFailed  = 7002
	ArticleDeleteFailed = 7003

	// 文章分类管理
	CatCreateFailed = 8001
	CatQueryFailed  = 8002
	CatUpdateFailed = 8003
	CatDeleteFailed = 8004

	// 页面管理
	PageCreateFailed = 9001
	PageQueryFailed  = 9002
	PageEditFailed   = 9003
	PageDeleteFailed = 9004

	// 评论管理
	CreateQueryFailed   = 10001
	GetQueryFailed      = 10002
	SendCommentFailed   = 10003
	SelectCommentFailed = 10004
	DeleteCommentFailed = 1005

	// findme 管理
	CreateFindMeFailed = 11001
	FindMeQueryFailed  = 11002
	FindMeModifyFailed = 11003
	DeleteFindMeFailed = 11004

	// 导航栏管理
	CreateNavFailed = 12001
	NavQueryFailed  = 12002
	NavEditFailed   = 12003
	NavDeleteFailed = 12004

	// 友链管理
	CreateFriendLinkFailed = 13001
	FriendLinkQueryFailed  = 13002
	FriendLinkModifyFailed = 13003
	FriendLinkDeleteFailed = 13004
)

var codeMsg = map[int]string{
	SUCCESS: "ok",
	ERROR:   "error",

	// JWT中间件
	TokenNotExist:       "token not exist",
	TokenInvalid:        "token invalid",
	TokenParseFailed:    "failed to parse claims",
	TokenGenerateFailed: "token generate failed",

	// 2FA
	GenerateSecretFailed: "generate secret failed",
	CodeError:            "code error",

	// QRcode
	GenerateQRFailed: "generate qrcode failed",

	// 管理员用户
	UserInfoError:       "user information error",
	UserExist:           "user exist",
	UserNotExist:        "user not exist",
	UserNotValidate:     "user not validate",
	CheckUserFailed:     "check user failed",
	LoginError:          "an error occurred while logging in",
	UidError:            "uid error",
	PasswordError:       "password error",
	CantLogin:           "this ipdress can't login agin",
	ResetPasswordFailed: "reset password failed",
	QueryUserFailed:     "query user failed",

	// 邮件发送
	SendEmailFailed:   "email send failed",
	ValidateCodeError: "validate coede error",

	// 博客设置
	BlogNotSet:    "blog not set",
	BlogSetFailed: "blog set failed",

	// 文章管理
	ArticleUpdateFailed: "article update failed",
	ArticleQueryFailed:  "article query failed",
	ArticleDeleteFailed: "article delete failed",

	// 文章分类管理
	CatCreateFailed: "category create failed",
	CatQueryFailed:  "category query failed",
	CatUpdateFailed: "category update failed",
	CatDeleteFailed: "category delete failed",

	// 页面管理
	PageCreateFailed: "page create failed",
	PageQueryFailed:  "page query failed",
	PageEditFailed:   "page edit failed",
	PageDeleteFailed: "page delete failed",

	// 评论管理
	CreateQueryFailed:   "create query failed",
	GetQueryFailed:      "get query failed",
	SendCommentFailed:   "send comment failed",
	SelectCommentFailed: "select comment failed",
	DeleteCommentFailed: "delete comment failed",

	// findme 管理
	CreateFindMeFailed: "create findme failed",
	FindMeQueryFailed:  "findme query failed",
	FindMeModifyFailed: "findme modify failed",
	DeleteFindMeFailed: "delete findme failed",
	CreateNavFailed:    "create navigation failed",
	NavQueryFailed:     "navigation query failed",
	NavEditFailed:      "navigation edit failed",
	NavDeleteFailed:    "navigation delete failed",

	// 友链管理
	CreateFriendLinkFailed: "create friend link failed",
	FriendLinkQueryFailed:  "friend link query failed",
	FriendLinkModifyFailed: "friend link modify failed",
	FriendLinkDeleteFailed: "friend link delete failed",
}

func GetErrorMsg(code int) string {
	return codeMsg[code]
}

File: nyalog-SSR-docker/vue-client/src/modules/errmsg.js
import MessageAPI from '../components/message.vue'

function errmsg(code) {
    switch (code) {
        case 1001:
            window.$message.error('Token不存在');
            break;
        case 1002:
            window.$message.error('Token非法');
            break;
        case 1003:
            window.$message.error('Token解析失败');
            break;
        case 1004:
            window.$message.error('Token产生失败');
            break;
        case 3001:
            window.$message.error('2FA生成失败');
            break;
        case 3002:
            window.$message.error('2FA验证码错误');
            break;
        case 4001:
            window.$message.error('二维码生成失败');
            break;
        case 2001:
            window.$message.error('用户信息错误');
            break;
        case 2002:
            window.$message.error('用户已存在');
            break;
        case 2003:
            window.$message.error('用户不存在');
            break;
        case 2004:
            window.$message.error('用户未验证');
            break;
        case 2005:
            window.$message.error('用户检查失败');
            break;
        case 2006:
            window.$message.error('用户登录失败');
            break;
        case 2007:
            window.$message.error('用户UID错误');
            break;
        case 2008:
            window.$message.error('密码错误');
            break;
        case 2009:
            window.$message.error('IP限制');
            break;
        case 2010:
            window.$message.error('重置密码失败');
            break;
        case 2011:
            window.$message.error('查询用户失败');
            break;
        case 2012:
            window.$message.error('用户信息更改失败');
            break;
        case 5001:
            window.$message.error('邮件验证码发送失败');
            break;
        case 5002:
            window.$message.error('验证邮件验证码失败');
            break;
        case 6001:
            window.$message.error('博客信息未设置');
            break;
        case 6002:
            window.$message.error('博客信息设置失败');
            break;
        case 6003:
            window.$message.error('查询博客信息失败');
            break;
        case 7001:
            window.$message.error('文章更新失败');
            break;
        case 7002:
            window.$message.error('文章查询失败');
            break;
        case 7003:
            window.$message.error('文章删除失败');
            break;
        case 8001:
            window.$message.error('分类创建失败');
            break;
        case 8002:
            window.$message.error('分类查询失败');
            break;
        case 8003:
            window.$message.error('分类更新失败');
            break;
        case 8004:
            window.$message.error('分类删除失败');
            break;
        case 9001:
            window.$message.error('页面创建失败');
            break;
        case 9002:
            window.$message.error('页面查询失败');
            break;
        case 9003:
            window.$message.error('页面编辑失败');
            break;
        case 9004:
            window.$message.error('页面删除失败');
            break;
        case 10001:
            window.$message.error('创建登录信息失败');
            break;
        case 10002:
            window.$message.error('登录信息失败');
            break;
        case 10003:
            window.$message.error('发送评论失败');
            break;
        case 10004:
            window.$message.error('查询评论失败');
            break;
        case 10005:
            window.$message.error('删除评论失败');
            break;
        case 11001:
            window.$message.error('新增findme失败');
            break;
        case 11002:
            window.$message.error('查询findme失败');
            break;
        case 11003:
            window.$message.error('修改findme失败');
            break;
        case 11004:
            window.$message.error('删除findme失败');
            break;
        case 12001:
            window.$message.error('创建导航栏元素失败');
            break;
        case 12002:
            window.$message.error('查询导航栏元素失败');
            break;
        case 12003:
            window.$message.error('编辑导航栏元素失败');
            break;
        case 12004:
            window.$message.error('删除导航栏元素失败');
            break;
        case 13001:
            window.$message.error('创建友链失败');
            break;
        case 13002:
            window.$message.error('查询友链失败');
            break;
        case 13003:
            window.$message.error('修改友链失败');
            break;
        case 13004:
            window.$message.error('删除友链失败');
            break;
        default:
            window.$message.error('未知错误');
            break;
    }
}

export default errmsg